React_Day6/avengers/src/components/BodyComponent.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+
+import CardComponent from "./CardComponent";
+import SearchBarComponent from "./SearchBarComponent";
+import NoResultFound from "./NoResultFound";
+import fetchService from "../utils/apiCall.js";
+
+function BodyComponent() {
+
+  const fetchedData = fetchService();
+  const [avengersData, setAvengersData] = useState([]);
+  const [afterFilterData, setAfterFilterData] = useState([]);
+
+  useEffect(() => {
+    fetchedData.then((res) => {
+      const data = [].concat(...res);
+      setAvengersData(data);
+      setAfterFilterData(data);
+    });
+  },[]);
+
+  return (
+    <>
+      <SearchBarComponent
+        avengersData={avengersData}
+        setAfterFilterData={setAfterFilterData}
+      />
+      {afterFilterData.length > 0 ? (
+        <div id="bodylayout" className="bodylayout-class">
+          {afterFilterData.map((i) => (

```

Comment:
instead of i can give the name something more meaningful. i gives the feeling of index